core/src/main/java/org/elasticsearch/index/query/GeoPolygonQueryBuilder.java

```diff
@@ -168,53 +190,58 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
             throw new QueryShardException(context, "field [" + fieldName + "] is not a geo_point field");
         }
 
-        List<GeoPoint> shell = new ArrayList<GeoPoint>();
-        for (GeoPoint geoPoint : this.shell) {
-            shell.add(new GeoPoint(geoPoint));
-        }
-        final int shellSize = shell.size();
-
-        // validation was not available prior to 2.x, so to support bwc
-        // percolation queries we only ignore_malformed on 2.x created indexes
-        if (!GeoValidationMethod.isIgnoreMalformed(validationMethod)) {
-            for (GeoPoint point : shell) {
-                if (!GeoUtils.isValidLatitude(point.lat())) {
-                    throw new QueryShardException(context, "illegal latitude value [{}] for [{}]", point.lat(),
-                            GeoPolygonQueryBuilder.NAME);
+        List<List<List<GeoPoint>>> polys = new ArrayList<>(this.polygons);
+        double[] shellLat = null, lat;
+        double[] shellLon = null, lon;
```

Comment:
Hmm, `lat` and `lon` are not used in this outer scope, only down inside the nested `for` loop?  Can you move their declaration (the `double[]`) down there?